src/shims/unix/fs.rs

```diff
@@ -1086,8 +1088,18 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 let name_bytes = name.as_encoded_bytes();
                 let name_len = u64::try_from(name_bytes.len()).unwrap();
 
-                let dirent64_layout = this.libc_ty_layout("dirent64");
-                let d_name_offset = dirent64_layout.fields.offset(4 /* d_name */).bytes();
+                let is_linux = matches!(&*this.tcx.sess.target.os, "linux");
+
+                let dirent64_layout = if is_linux {
+                    this.libc_ty_layout("dirent64")
```

Comment:
So Solaris has a readdir64 function that takes an argument of type dirent (not dirent64)? How does this make sense...?

In the libc crate I cannot find a `readdir64` for Solarish at all.

Reply:
no there is no readdir64 on solaris but because of [this alias ](https://github.com/rust-lang/rust/blob/c94848c046d29f9a80c09aae758e27e418a289f2/library/std/src/sys/pal/unix/fs.rs#L33) I got readdir64 in the unsupported call error before.

Reply:
> because of [this alias ](https://github.com/rust-lang/rust/blob/c94848c046d29f9a80c09aae758e27e418a289f2/library/std/src/sys/pal/unix/fs.rs#L33)I got readdir64 as symbol name in the unsupported call error before.

That can't be right, such an alias does not change the symbol name.

Please remove the readdir64 on Solarish then if it does not exist there.

Reply:
Seems locally I no longer need to add readdir64. Glad that's the case.. Let's see CI then I ll rephrase a bit the original commit message.